src/test/java/com/productdock/library/inventory/producer/KafkaProducerTest.java

```diff
@@ -0,0 +1,75 @@
+package com.productdock.library.inventory.producer;
+
+import com.productdock.library.inventory.book.BookAvailabilityMessage;
+import com.productdock.library.inventory.book.BookRepository;
+import com.productdock.library.inventory.data.provider.KafkaTestBase;
+import com.productdock.library.inventory.record.RentalRecord;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.time.Duration;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+import static com.productdock.library.inventory.data.provider.BookMother.defaultBookAvailabilityMessage;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+@SpringBootTest
+class KafkaProducerTest extends KafkaTestBase {
+
+    public static final String FIRST_BOOK = "1";
+    public static final int AVAILABLE_BOOK_COUNT = 1;
+    public static final String TEST_FILE = "testRecord.txt";
```

Comment:
Test file testRecord.txt used here shouldn't exist after tests are finished, make sure it is deleted before pushing to remote